internal/discovery/discovery.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2023 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package discovery provides a discovery and resolution logic for GVKs.
+package discovery
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// Interval is the interval at which the discovery client fetches the list of objects.
+const Interval = 30 * time.Second
+
+// Multiplier is the multiplier for the discovery interval, and the upper bound till we check for resolved GVKs to be populated.
```

Comment:
Can we have a `const Timeout = 180 * time.Second` here instead of doing math in the code?

Reply:
There's [one instance](https://github.com/kubernetes/kube-state-metrics/pull/1851/files#diff-dc8fe36bd1edf1b4cd03bacbd94b02cd4226717fe7e1a6474c534f5b5db30227R376) where we are only using `Interval`, so multiplying them would mean a division there.

Reply:
We could have two variables: Interval and Timeout